app/web/features/search/SearchBox.tsx

```diff
@@ -55,42 +55,39 @@ export default function SearchBox({
   const { t } = useTranslation([GLOBAL, SEARCH]);
   const classes = useStyles();
 
-  const { control, setValue, errors, register, handleSubmit } = useForm({
+  const { control, errors } = useForm({
     mode: "onChange",
   });
 
-  function handleOnChangeAutocomplete(event: "" | GeocodeResult) {
+  const handleOnChangeAutocomplete = (event: GeocodeResult) => {
     if (event) {
       setLocationResult(event);
     }
-  }
+  };
 
-  function handleOnChangeKeyword(event: React.ChangeEvent<HTMLInputElement>) {
+  const handleOnChangeKeyword = (
+    event: React.ChangeEvent<HTMLInputElement>
+  ) => {
     setQueryName(event.target.value);
-    setLocationResult({
-      ...locationResult,
-      location: undefined,
-    });
-  }
+    setLocationResult(undefined);
+  };
 
-  function handleOnChangeRadioButton(event: React.ChangeEvent, value: string) {
-    setSearchType(value as "location" | "keyword");
-    setLocationResult({
-      ...locationResult,
-      name: "",
-      simplifiedName: "",
-      location: undefined,
-    });
-    setQueryName(undefined);
-  }
+  const handleOnChangeRadioButton = (
+    event: React.ChangeEvent<HTMLInputElement>,
+    value: "location" | "keyword"
+  ) => {
+    setSearchType(value);
+    setLocationResult(undefined);
```

Comment:
I don't think this is what I wanted, I think I wanted to delete all the values of location but not bbox

Reply:
What behavior would that look like in the app? 

Since bbox represents a location, when would you want a bbox but not a location?

Reply:
Also on that note, do you know what triggers this particular function? I tried every form field in the modal, but nothing made this function run. I'm wondering if maybe it's old?

Reply:
it's the radio button on the form, available on desktop, basically let you choose between option a "location" and b "keyword" basically when user clicks keyword since the input of the location isn't defined I want to delete just the location, but keep the other values

Reply:
as I suspected, this gets broken with that change 😅 (🤷‍♂️) 
![image](https://github.com/user-attachments/assets/b0a1c59e-640d-4ab5-a33f-b525dc159c0b)

Reply:
don't worry, I make it work plus keepting the new types of this PR, I take care of this